### PR TITLE
[CARBONDATA-1213] Removed rowCountPercentage check and fixed IUD data load issue

### PR DESCRIPTION
**Pull request by manishgupta88**
Problems:
1. Row count percentage not required with high cardinality threshold check
2. IUD returning incorrect results in case of update on high cardinality column

Analysis:
1. In case a column is identified as high cardinality column still it is not getting converted to no dictionary column because of another parameter check called rowCountPercentage. Default value of rowCountPercentage is 80%. Due to this even though high cardinality column is identified, if it is less than 80% of the total number of rows it will be treated as dictionary column. This can still lead to executor lost failure due to memory constraints.
2. RLE on a column is not being set correctly and due to incorrect code design RLE applicable on a column is decided by a different part of code from the one which is actually applying the RLE on a column. Because of this Footer is getting filled with incorrect RLE information and query is failing.

Fix:
1. Remove an unwanted check for rowCountPercentage.
2. RLE applicability on a column should be decided from a common place in the code.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2657/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/86/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/572/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test' /><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/572/org.apache.carbondata$carbondata-spark-common-test/testReport'>carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/572/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.dataload/TestDataLoadWithColumnsMoreThanSchema/test_for_duplicate_column_name_in_the_Fileheader_options_in_load_command/'><strong>org.apache.carbondata.spark.testsuite.dataload.TestDataLoadWithColumnsMoreThanSchema.test for duplicate column name in the Fileheader options in load command</strong></a></li></ul>

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/87/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2658/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/575/

File: processing/src/main/java/org/apache/carbondata/processing/store/TablePageEncoder.java [hunk 1]
```diff
@@ -149,29 +151,26 @@ private void encodeAndCompressDimensions(TablePage tablePage, EncodedData encode
...
           // timestamp and date column
-          indexStorages[indexStorageOffset] =
-              encodeAndCompressDirectDictDimension(
-                  tablePage.getDictDimensionPage()[++dictionaryColumnCount].getByteArrayPage(),
-                  isSortColumn,
-                  isUseInvertedIndex[i] & isSortColumn);
+          indexStorages[indexStorageOffset] = encodeAndCompressDirectDictDimension(
+              tablePage.getDictDimensionPage()[++dictionaryColumnCount].getByteArrayPage(),
+              isSortColumn, isUseInvertedIndex[i] & isSortColumn,
+              CarbonDataProcessorUtil.isRleApplicableForColumn(DimensionType.DIRECT_DICTIONARY));
```
Comment (ravipesala) [inline processing/src/main/java/org/apache/carbondata/processing/store/TablePageEncoder.java]: pass `dimensionSpec.getType(i)` to `CarbonDataProcessorUtil.isRleApplicableForColumn`

File: processing/src/main/java/org/apache/carbondata/processing/store/TablePageEncoder.java [hunk 2]
```diff
@@ -149,29 +151,26 @@ private void encodeAndCompressDimensions(TablePage tablePage, EncodedData encode
...
           // high cardinality dimension, encoded as plain string
-          indexStorages[indexStorageOffset] =
-              encodeAndCompressNoDictDimension(
-                  tablePage.getNoDictDimensionPage()[++noDictionaryColumnCount].getByteArrayPage(),
-                  isSortColumn,
-                  isUseInvertedIndex[i] & isSortColumn);
+          indexStorages[indexStorageOffset] = encodeAndCompressNoDictDimension(
+              tablePage.getNoDictDimensionPage()[++noDictionaryColumnCount].getByteArrayPage(),
+              isSortColumn, isUseInvertedIndex[i] & isSortColumn,
+              CarbonDataProcessorUtil.isRleApplicableForColumn(DimensionType.PLAIN_VALUE));
```
Comment (ravipesala) [inline processing/src/main/java/org/apache/carbondata/processing/store/TablePageEncoder.java]: pass `dimensionSpec.getType(i)` to `CarbonDataProcessorUtil.isRleApplicableForColumn`

**Comment by manishgupta88** [comment]
@ravipesala ...handled review comments...kindly review

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/92/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2664/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/584/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**